crates/oxc_transformer/src/es2015/new_target.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{context::TransformerCtx, TransformOptions, TransformTarget};
+use oxc_allocator::Vec;
+use oxc_ast::{ast::*, AstBuilder, AstKind, VisitMut};
+use oxc_diagnostics::miette;
+use oxc_span::{Atom, Span, SPAN};
+use oxc_syntax::operator::BinaryOperator;
+use std::rc::Rc;
+
+/// ES2015: New Target
+///
+/// References:
+/// * <https://babel.dev/docs/babel-plugin-transform-template-new-target>
+/// * <https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-new-target>
+pub struct NewTarget<'a> {
+    ast: Rc<AstBuilder<'a>>,
+    ctx: TransformerCtx<'a>,
+    kinds: Vec<'a, NewTargetKind>,
+}
+
+#[derive(Debug)]
+enum NewTargetKind {
+    Method,
+    Constructor,
+    Function(Option<Atom>),
+}
+
+impl<'a> VisitMut<'a> for NewTarget<'a> {
+    fn enter_node(&mut self, kind: AstKind<'a>) {
```

Comment:
Ideally, we don't need to store kind, we can get the kind from `ctx.semantic().nodes().get()`. 
However, there is currently an issue where `ctx.nodes()` does not correspond exactly to our `ast`. Because `ast` is mutable